doc/mgmt/SONiC_DNS_Support.md

```diff
@@ -0,0 +1,433 @@
+- OpenConfig YANG model for DNS openconfig-system.yang   
+- SONiC DNS model for DNS based on Redis DB schema   
+- transformer functions to    
+   * convert OpenConfig YANG model to SONiC YANG model for DNS related configurations   
+
+### 1.2.3 SAI Overview
+
+# 2 Functionality
+
+## 2.1 Target Deployment Use Cases
+Manage/configure DNS configurations via gNMI, REST and CLI interfaces
+
+## 2.2 Functional Description
+Provide CLI, gNMI and REST support for DNS configurations.
+
+## 2.3 Backend change to support new configurations
+Provide change in management framework and hostcfgd modules.
+
+# 3 Design
+
+## 3.1 Overview
+
+Enhancing the management framework backend code and transformer methods to add support for DNS.
+
+## 3.2 DB Changes
+
+### 3.2.1 CONFIG DB
+```
+; DNS configuration attributes global to the system. Only one instance of the table exists in the system.
+; Key
+global_key           = “global”  ;  TACACS+ global configuration
+; Attributes
+src_intf               = ifname  ;  source interface address for the outgoing DNS packets
+
+```
+
+```
+; DNS name server configuration in the system.
+; Key
+server_key           = IPAddress  ;  DNS server’s address
+; Attributes
+No attributes are introduced as part of this design.
+```
+
+### 3.2.2 APP DB
+
+### 3.2.3 STATE DB
+
+### 3.2.4 ASIC DB
+
+### 3.2.5 COUNTER DB
+
+## 3.3 Switch State Service Design
+
+### 3.3.1 Orchestration Agent
+
+### 3.3.2 Other Process
+
+## 3.4 SyncD
+
+## 3.5 SAI
+
+## 3.6 User Interface
+
+### 3.6.1 Data Models
+
+YANG models needed for DNS handling in the management framework:
+1. **openconfig-system.yang**  
+
+2. **sonic-system-dns.yang**
+
+Supported yang objects and attributes:
+```diff
+
+module: openconfig-system
+      +--rw system
+
+      +--rw dns
+           |  +--rw config
+           |  |  +--rw search*                      oc-inet:domain-name
+           |  |  +--rw oc-sys-ext:source-intf?   -> /oc-if:interfaces/interface/name
+           |  +--ro state
+           |  |  +--ro search*                      oc-inet:domain-name
+           |  |  +--ro oc-sys-ext:source-intf?   -> /oc-if:interfaces/interface/name
+           |  +--rw servers
+           |  |  +--rw server* [address]
+           |  |     +--rw address    -> ../config/address
+           |  |     +--rw config
+           |  |     |  +--rw address?   oc-inet:ip-address
+           |  |     |  +--rw port?      oc-inet:port-number
+           |  |     +--ro state
+           |  |        +--ro address?   oc-inet:ip-address
+           |  |        +--ro port?      oc-inet:port-number
+
+The above "source-intf" & "address" fields are supported in the scope of this work.
+
+module: sonic-system-dns
+  +--rw sonic-system-dns
+     +--rw DNS_SERVER
+     |  +--rw DNS_SERVER_LIST* [ipaddress]
+     |     +--rw ipaddress    inet:ip-address
+     +--rw DNS
+        +--rw DNS_LIST* [type]
+           +--rw type      enumeration ('global')
+           +--rw src_intf?   union
+
+```

```

Comment:
From offline comments, I assume there is no plan to support following n upcoming B+ release?
Is there a way to specify the domain name (domain option of resolv.conf)?
Is there a way to  specify search list for host-name lookup (search option)?

Reply:
Yes, no plan to support these commands for 3.1.